modules/exploits/linux/local/rc_local_persistence.rb

```diff
@@ -0,0 +1,57 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Post::Unix
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'rc.local Persistence',
+      'Description'    => %q(
+        This module will edit /etc/rc.local in order to persist a payload.
+      ),
+      'License'        => MSF_LICENSE,
+      'Author'         => [ 'Eliott Teissonniere' ],
+      'Platform'       => [ 'unix', 'linux' ],
+      'Arch'           => ARCH_CMD,
+      'Payload'        => {
+        'BadChars'   => '#%\n',
```

Comment:
Is this meant to be `\n`, as in newline? Ruby does not permit escaping within single quotes.

```
2.3.0 :001 > puts 'asdf\nzxcv'
asdf\nzxcv
 => nil 
2.3.0 :002 > puts "asdf\nzxcv"
asdf
zxcv
 => nil 
```

Using `\x0a` is preferred

Reply:
Ah oops, will put that in double quotes, feels cleaner IMO